panel/depends.py

```diff
@@ -26,7 +26,78 @@ def param_value_if_widget(arg):
 
 
 def depends(*args, **kwargs):
+    """
+    Annotates a function or Parameterized method to express its
+    dependencies on a set of parameters. This allows libraries
+    such as Panel to watch those dependencies for changes and update
+    the output of the annotated function when one of the dependencies
+    changes.
+
+    The specified dependencies can be either be Parameter instances,
+    widgets, or, if a method is supplied, they can be defined
+    as strings referring to Parameters of the class, or Parameters of
+    subobjects (Parameterized objects that are values of this object's
+    parameters).
```

Comment:
```suggestion
    Python decorator annotating a function or Parameterized method to 
    express its dependencies on a set of Parameters. This declaration
    allows libraries like Panel to watch those dependencies for changes
    and invoke the function or method to get an updated output when one
    of the dependencies changes.

    This function is the same as the corresponding param.depends 
    decorator, but extended so that if widgets are provided, the underlying
    value parameter of the widget is extracted as a dependency. This 
    extension is solely for syntactic convenience, allowing the widget to be
    passed in as a synonym for the underlying parameter, leaving this 
    decorator otherwise behaving the same as the underlying Param
    depends decorator. 
    
    For the Panel version of the decorator, the specified dependencies can
    either be Parameter instances, Panel or ipywidgets widgets, or, if a 
    Parameterized method is supplied rather than a function, they can be
    defined as strings referring to Parameters of that class, or Parameter 
    objects of that object's subobjects (i.e., Parameterized objects that are
    values of this object's parameters).
```